parse.go

```diff
@@ -108,7 +108,14 @@ type ParseOptions struct {
 	// encountered by the parser to determine the valid contexts and argument count of the
 	// directive. Set this option to enable parsing of directives belonging to non-core or
 	// dynamic NGINX modules that follow the usual grammar rules of an NGINX configuration.
+	// Deprecated: use MatchAny instead
```

Comment:
If the options aren't going to be "validated" to ensure the caller is not mixing the two options, then I think the behavior that these would be ignored should at least be documented.

Reply:
Sure! Added it in latest commit